### ISSUE TITLE
SIO_LE_NATIVE, SIO_PLAY undefined ???

### ISSUE THREAD

**Issue by jiapei100** [erikd/libsndfile#444]
```
[ 90%] Building C object CMakeFiles/sndfile-play.dir/programs/sndfile-play.c.o
/usr/bin/cc  -I....../libsndfile/build/src -I....../libsndfile/src  -O3 -DNDEBUG   -std=gnu99 -o CMakeFiles/sndfile-play.dir/programs/sndfile-play.c.o   -c ....../libsndfile/programs/sndfile-play.c
....../libsndfile/programs/sndfile-play.c: In function ‘sndio_play’:
....../libsndfile/programs/sndfile-play.c:674:17: error: storage size of ‘par’ isn’t known
  struct sio_par par ;
                 ^~~
....../libsndfile/programs/sndfile-play.c:692:14: warning: implicit declaration of function ‘sio_open’; did you mean ‘sf_open’? [-Wimplicit-function-declaration]
   if ((hdl = sio_open (NULL, SIO_PLAY, 0)) == NULL)
              ^~~~~~~~
              sf_open
....../libsndfile/programs/sndfile-play.c:692:30: error: ‘SIO_PLAY’ undeclared (first use in this function)
   if ((hdl = sio_open (NULL, SIO_PLAY, 0)) == NULL)
                              ^~~~~~~~
....../libsndfile/programs/sndfile-play.c:692:30: note: each undeclared identifier is reported only once for each function it appears in
....../libsndfile/programs/sndfile-play.c:697:3: warning: implicit declaration of function ‘sio_initpar’ [-Wimplicit-function-declaration]
   sio_initpar (&par) ;
   ^~~~~~~~~~~
....../libsndfile/programs/sndfile-play.c:702:12: error: ‘SIO_LE_NATIVE’ undeclared (first use in this function)
   par.le = SIO_LE_NATIVE ;
            ^~~~~~~~~~~~~
....../libsndfile/programs/sndfile-play.c:704:9: warning: implicit declaration of function ‘sio_setpar’ [-Wimplicit-function-declaration]
   if (! sio_setpar (hdl, &par) || ! sio_getpar (hdl, &par))
         ^~~~~~~~~~
....../libsndfile/programs/sndfile-play.c:704:37: warning: implicit declaration of function ‘sio_getpar’ [-Wimplicit-function-declaration]
   if (! sio_setpar (hdl, &par) || ! sio_getpar (hdl, &par))
                                     ^~~~~~~~~~
....../libsndfile/programs/sndfile-play.c:709:9: warning: implicit declaration of function ‘sio_start’; did you mean ‘va_start’? [-Wimplicit-function-declaration]
   if (! sio_start (hdl))
         ^~~~~~~~~
         va_start
....../libsndfile/programs/sndfile-play.c:715:4: warning: implicit declaration of function ‘sio_write’; did you mean ‘pwrite’? [-Wimplicit-function-declaration]
    sio_write (hdl, buffer, readcount * sizeof (short)) ;
    ^~~~~~~~~
    pwrite
....../libsndfile/programs/sndfile-play.c:717:3: warning: implicit declaration of function ‘sio_close’; did you mean ‘sf_close’? [-Wimplicit-function-declaration]
   sio_close (hdl) ;
   ^~~~~~~~~
   sf_close
CMakeFiles/sndfile-play.dir/build.make:65: recipe for target 'CMakeFiles/sndfile-play.dir/programs/sndfile-play.c.o' failed
make[2]: *** [CMakeFiles/sndfile-play.dir/programs/sndfile-play.c.o] Error 1
make[2]: Leaving directory '....../libsndfile/build'
CMakeFiles/Makefile2:551: recipe for target 'CMakeFiles/sndfile-play.dir/all' failed
make[1]: *** [CMakeFiles/sndfile-play.dir/all] Error 2
make[1]: Leaving directory '....../libsndfile/build'
Makefile:132: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by evpobr** [comment]
Hello, i need more information:

* OS
* `config.h`
* Output of configutation process

**Comment by jiapei100** [comment]
Solved... it looks like sndio.h is missing now...
There is NO definition about

```
/*
 * mode bitmap
 */
#define SIO_PLAY	1
#define SIO_REC		2
#define MIO_OUT		4
#define MIO_IN		8
```

and

```
#if BYTE_ORDER == LITTLE_ENDIAN
#define SIO_LE_NATIVE 1
#else
#define SIO_LE_NATIVE 0
#endif
```


Cheers
Pei

**Comment by mipac** [comment]
same issue, on debian 9
what is the patch or workaround?

**Comment by erikd** [comment]
@mipac  That error message should only occur on OpenBSD. If you are seeing that on Debian you are doing something very wrong.

**Comment by mipac** [comment]
lib sndio was not installed... I solved with :

`apt-get install libsndio-dev sndio-tools`

sorry for the noise, hope that will help someone ;)

**Comment by marc1uk** [comment]
Also running debian 9, also had this problem. I installed the packages recommended in the README and those remarked by mipac, but still no success.
Going from what @erikd said, I took a look at `sndfile_play.c` and it looks like `sndio.h` is `#included` only if `__linux__` is found, but the failing function `sndio_play` is built if `HAVE_SNDIO_H` is defined - regardless of the setting of `__linux__`.
I can't follow the CMake trails well enough to know, but I wonder if there is an issue in the `CMakeLists` file that is setting `HAVE_SNDIO_H` to true on linux platforms, when it should not be?
I replaced the CMakeListst line `#set (HAVE_SNDIO_H ${SNDIO_FOUND})` with `set (HAVE_SNDIO_H FALSE)` and the build completed.
